torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -771,8 +771,9 @@ void enableProfiler(
   KinetoThreadLocalState::push(state_ptr);
 
   if (has_cpu) {
-    config.global() ? pushProfilingCallbacks</*global=*/true>(scopes)
-                    : pushProfilingCallbacks</*global=*/false>(scopes);
+    config.global() || config.experimental_config.profile_all_threads
```

Comment:
config.global() is used in more situations that shouldn't necessarily be affected by this experimental config, but we can have a separate function inside `config` to avoid those ors

Reply:
Oh right, good call. We use the `global` for control flow to separate on-demand from auto-trace. In an ideal world, I think `global` should be used for global callbacks and we have some other function for on-demand vs auto-trace. I can work on cleaning that up in a future PR (would just be renaming).